pyarn/lockfile.py

```diff
@@ -148,6 +149,33 @@ def _quote_key_if_needed(key):
     return ", ".join(f'"{k}"' if _needs_quoting(k) else k for k in keys)
 
 
+def _check_for_quoted_comma(lockfile_str):
+    for line in lockfile_str.split('\n'):
+        if line and not line[0].isspace() and line[0] != '#':
+            quoted_comma = _extract_quoted_comma(line)
+            if quoted_comma:
+                raise ValueError(f"Following key has a quoted comma: {quoted_comma}")
+
+
+def _extract_quoted_comma(key):
+    # Finds and returns the first quoted comma in the key; returns None if it does not exist
+    if not key:
+        return False
+
+    start = -1
+    found = False
+    for i, c in enumerate(key):
+        if start == -1 and c == '"':
+            start = i
+        elif start != -1 and c == '"':
+            if found:
+                return key[start:i + 1]
+            start = -1
+        elif start != -1 and c == ',':
+            found = True
+    return None
```

Comment:
Would it be possible to incorporate the check into the parsing code itself? I feel like this adds a tiny but completely separate bit of parsing logic.

Reply:
In parser.py, the input parameter p for all the functions are always provided without quotations. If the key was originally `"foo, bar"`, then p will be `foo, bar`. So, there is no way to check for quoted commas in there.

Reply:
I'm not 100% sure about this, but maybe it could be handled in `p_list` https://github.com/containerbuildsystem/pyarn/blob/41fd2ddee844643c385dece803d415335756cfad/pyarn/parser.py#L63-L66

AFAIK this is the rule that parser comma-separated lists of keys, so perhaps checking that those keys don't contain commas would work?

Reply:
This only works for lines with multiple keys like `foo, "foo, bar"` but not for just `"foo, bar"`

Reply:
Also, by doing this, other tests with several keys in one line fail (when they are not supposed to).

Reply:
Oh right, if you have `foo, bar, baz`, you get a situation where `p[1] = 'foo, bar'` and `p[3] = 'baz'`. Doesn't feel impossible to solve though :thinking: 

Throwing out an idea: `p_list` should return a list, `p_title` should check if `p[1]` is a list or a string (and should check for commas in both cases)